internal/log/log_test.go

```diff
@@ -1,366 +1,71 @@
-package log
+package log_test
 
 import (
 	"bytes"
-	"errors"
-	"fmt"
-	"regexp"
+	"context"
+	"log/slog"
 	"strings"
 	"testing"
 
-	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
-	"go.uber.org/zap/zaptest/observer"
+	"github.com/go-spatial/tegola/internal/log"
 )
 
-const ExpectedFilename = "log_test.go"
-
-func TestSetLevel(t *testing.T) {
-
-	type tcase struct {
-		isDefault bool
-		lvl       Level
-		tst       func() bool
-	}
-	fn := func(tc tcase) (string, func(*testing.T)) {
-
-		name := "Default"
-		if !tc.isDefault {
-			name = tc.lvl.String()
-		}
-
-		return name, func(t *testing.T) {
-
-			if !tc.isDefault {
-				l := level
-				SetLogLevel(tc.lvl)
-				// Restore the level back
-				defer SetLogLevel(l)
+func TestStackTraceHandler(t *testing.T) {
+	testCases := []struct {
+		name        string
+		level       slog.Level
+		message     string
+		expectStack bool
+	}{
+		{name: "Debug", level: slog.LevelDebug, message: "debug", expectStack: false},
+		{name: "Info", level: slog.LevelInfo, message: "info", expectStack: false},
+		{name: "Warn", level: slog.LevelWarn, message: "warn", expectStack: false},
+		{name: "Error", level: slog.LevelError, message: "error", expectStack: true},
+		{name: "Silent", level: log.LevelSilent, message: "error", expectStack: false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var buf bytes.Buffer
+			baseHandler := slog.NewTextHandler(&buf, &slog.HandlerOptions{
+				Level:     slog.LevelDebug,
+				AddSource: true,
+			})
+			handler := log.NewHandler(baseHandler)
+			logger := slog.New(handler).WithGroup("testGroup").With("foo", "bar")
```

Comment:
I'm sorry I just noticed this. 
This function breaks our convention on how test should be written: (I thought I had it documented somewhere, but I can not find it. Will have to write up a document.)

```

func Test{{Name}}(t *testing.T) {

     type tcase struct {
       {{ fields for the testcase }}
     }

     fn := func(tc tcase)(func(*testing.T){
   // or ver(b)
  // fn := func(tc tcase) (string, func(*testing.T){ // this sig is hardly used
     return func(t *testing.T){
       {{ test body here }}
    }})

     tests := map[string]tcase{
       "test_1" : { ...  },
         .
         .
         .
      }

     for name, tc := range tests {
         t.Run(name, fn(tc))
     }
```

I would like to keep all our tests in the same structure.

}